libs/remix-ui/run-tab/src/lib/actions/events.ts

```diff
@@ -33,7 +37,6 @@ export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
     }
     setFinalContext(plugin, dispatch)
     fillAccountsList(plugin, dispatch)
-    await loadPinnedContracts(plugin, dispatch)
```

Comment:
what happens if one change the network in metamask (or any injected provider) ?

Reply:
> what happens if one change the network in metamask (or any injected provider) ?

I mean that the network get changed not by selecting a new env in udapp, but by selecting another network directly from metamask.

Reply:
both the cases are handled by this condition `if (currentNetwork.provider !== networkProvider() || (!isVM && currentNetwork.chainId !== network.id))` you can try it on netlify link